streaming/index.js

```diff
@@ -381,6 +381,23 @@ const startWorker = (workerId) => {
     attachCloseHandler(`${redisPrefix}${id}`, listener);
   };
 
+  const streamNotificationEnd = req => {
+    pgPool.connect((err, client, done) => {
+      if (err) {
+        log.error(err);
+        return;
+      }
+
+      client.query('UPDATE users SET last_read_notification_id = (SELECT id FROM notifications ORDER BY id DESC LIMIT 1) FROM oauth_access_tokens WHERE oauth_access_tokens.resource_owner_id = users.id AND oauth_access_tokens.token = $1 AND oauth_access_tokens.reading_notifications', [req.token], err => {
```

Comment:
I don't think performing UPDATEs through the streaming API is a good idea. First, the frequency with which the UPDATEs could be executed, and secondly because so far it was safe (and smart) to run the streaming API from PostgreSQL read-only replica, but with this change it wouldn't be possible.

At last: Streamed does not mean seen... E.g. delivered vs opened indicators in chat apps.

Reply:
Sorry for late reply. Do you have any alternative idea?